### PR TITLE
Per breakpoint (per row) plots

### PR DESCRIPTION
**Pull request by kgaonkar6**
## Description
The breakpoints are being plotted per Sample and FusionName selected as of yesterday's PR samplespecific_bpplots .
After discussing with @jharenza  we also want to plot specific breakpoint per Sample per FusionName.  I already had the feature for selecting breakpoints but it looks like I had them switched so editing that in this PR.

Which cane done in the core plot_breakpoints() as as follows :
left <- plot_breakpoints(sampleid = "BS_044XZ8ST", 
                         domainDataFrame = domainDataFrame, 
                         exons = exons, 
                         geneposition = "Left", 
                         fusionname = "ANTXR1--BRAF",
                        leftBreakpoint = "69152264")

In addition , I also a added the max_count limit to take the aggreagted fusionName count in plot_recurrent_genes and plot_recurrent_fusions.

## Request to reviewers
Please review the conditions to plot which are: 
If sample is not NULL plot for the specific sample
If leftBreakpoint/rightBreakpoint is not NULL then plot for specific breakpoint

## Changes affecting shiny_fuse
I guess for shiny_fuse the checkbox if checked should plot per breakpoint in the row @federicomarini can you add more information here if it's required?

### CODE REVIEW

**Comment by jharenza** [comment]
This seems to still show per sample plots for me, even when unchecking the box as default. Eg: `BS_044XZ8ST`

Screen Shot 2020-08-27 at 9 12 30 PM


@federicomarini can you make the default plot per row and then users can check if they want to see by sample?

**Review by jharenza — CHANGES_REQUESTED** [review]
see comment

**Comment by federicomarini** [comment]
> @federicomarini can you make the default plot per row and then users can check if they want to see by sample?
Sure!

**Comment by federicomarini** [comment]
> This seems to still show per sample plots for me, even when unchecking the box as default. Eg: BS_044XZ8ST

Can that be because there is only one sample anyway?
I did follow the path of the code via `message` commands and it seems to be correct.

**Comment by kgaonkar6** [comment]
Hi @federicomarini , to make it per row I think you'll have to provide the specific breakpoint to the function , accessing that per row as follows 

      fusion_for_content <- values$annofuse_tbl[row_id, "FusionName"]
      samplespec <- values$annofuse_tbl[row_id, "Sample"]
      right_breakpoint <-  values$annofuse_tbl[row_id, "RightBreakpoint"]

So then the following code will changed from this
https://github.com/d3b-center/annoFuse/blob/5fba25e109979bad54d5c3e4986106bb0a7c5c0f/R/shiny_fuse.R#L505-L519

to something like
```
if(input$bp_plot_samplespecific) {
       # plot all breakpoints in fusion per sample in row
        p <- plot_breakpoints(
          domainDataFrame = values$ann_domain,
          exons = values$data_exons,
          geneposition = "Right",
          sampleid = samplespec,
          fusionname = fusion_for_content,
        ) 
      } else { 
        # default plot per breakpoint in fusion per sample in row
        p <- plot_breakpoints(
          domainDataFrame = values$ann_domain,
          exons = values$data_exons,
          geneposition = "Right",
          fusionname = fusion_for_content,
          rightBreakpoint = right_breakpoint,
          sampleid = samplespec
        )
      }

```

File: R/shiny_fuse.R [hunk 1]
```diff
@@ -741,27 +741,30 @@ shiny_fuse <- function(out_annofuse = NULL) {
...
       })
     
     output$btn_dl_bpboth <- downloadHandler(
       filename = "annofuse_bpboth.pdf",
       content = function(file) {
-        ggsave(file, plot = values$plotobj_breakpoint_both #, 
-               # width = input$export_width,
-               # height = input$export_height, units = "cm"
+        ggsave(file, plot = values$plotobj_breakpoint_both, 
+               width = 20,
```
Comment (kgaonkar6) [inline R/shiny_fuse.R]: maybe longer width for both genes fused plots?
```suggestion
               width = 30,
```

**Review by kgaonkar6 — COMMENTED** [review]
buttons work like a charm!

I think all the points for breakpoint plots are being addressed in this PR.   @federicomarini and @jharenza  to approve changes

**Comment by jharenza** [comment]
looks good

**Pull request merged**